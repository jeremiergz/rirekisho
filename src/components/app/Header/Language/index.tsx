import Img, { FluidObject } from 'gatsby-image';
import React, { FunctionComponent } from 'react';
import Box from '../../../primitives/Box';
import FlexBox from '../../../primitives/FlexBox';
import Star from './Star';

const Language: FunctionComponent<LanguageProps> = ({ dense = false, img, item }) => {
  const maxStarsNumber = 5;
  const maxStarsDifference = maxStarsNumber - item.proficiencyLevel;
  const fullStarsNumber = [...Array(Math.floor(item.proficiencyLevel)).keys()];
  const halfStarsNumber = [...Array(Math.round(maxStarsDifference % 1)).keys()];
  const emptyStarsNumber = [...Array(Math.floor(maxStarsDifference)).keys()];
  return (
    <FlexBox alignItems="center" flexDirection="column" justifyContent="center" marginX={!dense && '12px'}>
      <Box width={{ _: 40, tablet: 48 }}>
        <Img fluid={img} imgStyle={{ borderRadius: 4, height: '100%', width: '100%' }} />
      </Box>
      <FlexBox marginTop={{ _: 0, tablet: 1 }}>
        {fullStarsNumber.map(fstar => (
          <Star height={{ _: 12, tablet: 16 }} key={fstar} type="full" />
        ))}
        {halfStarsNumber.map(fstar => (
          <Star height={{ _: 12, tablet: 16 }} key={fstar} type="half" />
        ))}
        {emptyStarsNumber.map(fstar => (
          <Star height={{ _: 12, tablet: 16 }} key={fstar} type="empty" />
        ))}
      </FlexBox>
    </FlexBox>
  );
};

Language.displayName = 'Language';

export type LanguageProps = {
  dense?: boolean;
  img: FluidObject;
  item: Models.Language;
};
export default Language;
