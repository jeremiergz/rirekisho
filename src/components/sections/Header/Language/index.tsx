import Box from '@primitives/Box';
import FlexBox from '@primitives/FlexBox';
import React from 'react';
import Star from './Star';

const Language: React.FC<LanguageProps> = ({ dense = false, img, item }) => {
  const maxStarsNumber = 5;
  const maxStarsDifference = maxStarsNumber - item.proficiencyLevel;
  const fullStarsNumber = [...Array(Math.floor(item.proficiencyLevel)).keys()];
  const halfStarsNumber = [...Array(Math.round(maxStarsDifference % 1)).keys()];
  const emptyStarsNumber = [...Array(Math.floor(maxStarsDifference)).keys()];
  return (
    <FlexBox alignItems="center" flexDirection="column" justifyContent="center" marginLeft={!dense && '12px'}>
      <Box width={{ _: 40, tablet: 48 }}>
        <Box alt={item.name} as="img" src={img} width="100%" />
      </Box>
      <FlexBox marginTop={{ _: 0, tablet: 1 }}>
        {fullStarsNumber.map(fstar => (
          <Star height={{ _: 12, tablet: 16 }} key={fstar} type="full" />
        ))}
        {halfStarsNumber.map(fstar => (
          <Star height={{ _: 12, tablet: 16 }} key={fstar} type="half" />
        ))}
        {emptyStarsNumber.map(fstar => (
          <Star height={{ _: 12, tablet: 16 }} key={fstar} type="empty" />
        ))}
      </FlexBox>
    </FlexBox>
  );
};

Language.displayName = 'Language';

export type LanguageProps = {
  dense?: boolean;
  img: string;
  item: Models.Language;
};
export default Language;
