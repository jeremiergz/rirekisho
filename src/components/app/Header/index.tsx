import { graphql, useStaticQuery } from 'gatsby';
import React, { FunctionComponent } from 'react';
import useTheme from '../../../hooks/useTheme';
import FlexBox from '../../primitives/FlexBox';
import Text from '../../primitives/Text';
import Language from './Language';

const Header: FunctionComponent = () => {
  const {
    flagImgNodes: { nodes: flagImgs },
    languageNodes: { nodes: languages },
  } = useStaticQuery<GraphQL.HeaderData>(graphql`
    query HeaderData {
      flagImgNodes: allImageSharp {
        nodes {
          fluid {
            originalName
            ...GatsbyImageSharpFluid
          }
        }
      }
      languageNodes: allLanguagesJson {
        nodes {
          img
          name
          proficiencyLevel
          sortOrder
        }
      }
    }
  `);
  const theme = useTheme();
  const flagsIndex = flagImgs.reduce((acc, flag) => {
    if (flag.fluid.originalName.includes('flag_')) acc[flag.fluid.originalName] = flag.fluid;
    return acc;
  }, {});
  return (
    <FlexBox
      alignItems="center"
      as="header"
      backgroundColor="primary"
      color="white"
      flexDirection={{ _: 'column', tablet: 'row' }}
      fontFamily={theme.fonts.title}
      height={{ _: 168, tablet: 96 }}
      justifyContent={{ _: 'center', tablet: 'space-between' }}
      paddingX={{ _: 3, tablet: 4, laptopS: 5 }}
    >
      <FlexBox alignItems="center" justifyContent={{ _: 'center', tablet: 'flex-start' }} paddingTop={2}>
        <Text
          fontSize={32}
          letterSpacing={{ _: -2, laptopL: 0 }}
          lineHeight="32px"
          textAlign="center"
          textTransform="uppercase"
          whiteSpace="nowrap"
        >
          Jeremie Rodriguez
        </Text>
      </FlexBox>
      <FlexBox marginBottom={{ _: 24, tablet: 0 }} marginTop={{ _: 2, tablet: 0 }}>
        {languages
          .sort((a, b) => a.sortOrder - b.sortOrder)
          .map((item, index) => (
            <Language
              dense={index === 0 || index === languages.length - 1}
              key={item.name}
              img={flagsIndex[item.img]}
              item={item}
            />
          ))}
      </FlexBox>
    </FlexBox>
  );
};

Header.displayName = 'Header';

export default Header;
