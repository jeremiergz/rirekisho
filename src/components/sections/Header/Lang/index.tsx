import Language from '@/models/Language';
import clsx from 'clsx';
import { GatsbyImage, getImage } from 'gatsby-plugin-image';
import React from 'react';
import Star from './Star';

function Lang({ dense = false, item }: LangProps): JSX.Element {
  const maxStarsNumber = 5;
  const maxStarsDifference = maxStarsNumber - item.proficiencyLevel;

  const fullStarsNumber = [...Array(Math.floor(item.proficiencyLevel)).keys()];
  const halfStarsNumber = [...Array(Math.round(maxStarsDifference % 1)).keys()];
  const emptyStarsNumber = [...Array(Math.floor(maxStarsDifference)).keys()];

  const image = getImage(item.img.src);

  return (
    <div className={clsx('flex flex-col items-center justify-center', !dense && 'ml-3')}>
      <div className="w-10 md:w-12">
        <GatsbyImage alt={item.name} image={image} />
      </div>
      <div className="mt-0.5 flex md:mt-1">
        {fullStarsNumber.map(fstar => (
          <Star className="h-3 md:h-4" key={fstar} type="full" />
        ))}
        {halfStarsNumber.map(fstar => (
          <Star className="h-3 md:h-4" key={fstar} type="half" />
        ))}
        {emptyStarsNumber.map(fstar => (
          <Star className="h-3 md:h-4" key={fstar} type="empty" />
        ))}
      </div>
    </div>
  );
}

export type LangProps = {
  dense?: boolean;
  item: Language;
};
export default Lang;
