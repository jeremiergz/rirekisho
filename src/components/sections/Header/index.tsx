import ThemeSwitch from '@/components/common/ThemeSwitch';
import useLanguagesData from '@/components/hooks/data/useLanguagesData';
import usePersonalDetailsData from '@/components/hooks/data/usePersonalDetailsData';
import clsx from 'clsx';
import React from 'react';
import Language from './Language';

function Header(): JSX.Element {
  const {
    languages: { nodes: languages },
  } = useLanguagesData();
  const {
    personalDetails: { fullName },
  } = usePersonalDetailsData();

  return (
    <header
      className={clsx(
        'box-border flex h-36 md:h-20 items-center justify-center relative',
        'bg-gradient-to-b from-primary to-secondary dark:from-primary dark:via-primary-dark dark:to-secondary-dark',
      )}
    >
      <div
        className={clsx(
          'flex items-center flex-col md:flex-row justify-center md:justify-between',
          'px-6 md:px-8 lg:px-10',
          'max-w-7xl w-full',
        )}
      >
        <div className="flex items-center justify-center md:justify-start">
          <span
            className={clsx(
              'pt-1.5 text-3xl whitespace-nowrap',
              'leading-8 font-title text-gray-100 tracking-tighter uppercase',
            )}
          >
            {fullName}
            <div
              className={clsx('absolute -bottom-3 flex left-auto justify-center right-auto max-w-7xl', 'print:hidden')}
            >
              <ThemeSwitch />
            </div>
          </span>
        </div>
        <div className="flex mb-6 md:mb-0 mt-2 md:mt-0">
          {languages.map((item, index) => (
            <Language dense={index === 0} key={item.name} item={item} />
          ))}
        </div>
      </div>
    </header>
  );
}

export default Header;
