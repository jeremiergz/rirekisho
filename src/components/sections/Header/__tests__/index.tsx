import ThemeProvider from 'components/providers/ThemeProvider';
import * as Gatsby from 'gatsby';
import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import Header from '..';
import mockData from './mock.json';

describe('components/sections/Header Test Suite', () => {
  beforeAll(() => {
    const graphql = jest.spyOn(Gatsby, 'graphql');
    graphql.mockImplementation(() => '');
    const useStaticQuery = jest.spyOn(Gatsby, 'useStaticQuery');
    useStaticQuery.mockImplementation(() => mockData);
  });

  it('matches snapshot', () => {
    const jsx = (
      <ThemeProvider>
        <Header />
      </ThemeProvider>
    );
    expect(ReactTestRenderer.create(jsx).toJSON()).toMatchSnapshot();
  });
});
