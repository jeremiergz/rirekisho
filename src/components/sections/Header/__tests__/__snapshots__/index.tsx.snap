// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/sections/Header Test Suite matches snapshot 1`] = `
Array [
  <div
    className="sc-AxjAm gwsCit"
    display={
      Object {
        "_": "none",
        "tablet": "inline-block",
      }
    }
  >
    <button
      className="sc-AxjAm sc-AxirZ klyYhV"
      onClick={[Function]}
    >
      <svg
        className="sc-AxiKw jLREbQ"
        fill="white"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
        />
      </svg>
    </button>
  </div>,
  <header
    className="sc-AxjAm emuCQU"
    color="white"
    display="flex"
    fontFamily="title"
    height={
      Object {
        "_": 168,
        "tablet": 96,
      }
    }
    width="100%"
  >
    <div
      className="sc-AxjAm ksZfWg"
      display="flex"
      width="100%"
    >
      <div
        className="sc-AxjAm hkypPr"
        display="flex"
      >
        <span
          className="sc-AxhCb folTfX"
          fontSize={32}
          letterSpacing={-2}
        >
          Jeremie Rodriguez
        </span>
      </div>
      <div
        className="sc-AxjAm enTfgo"
        display="flex"
      >
        <div
          className="sc-AxjAm ggaXpg"
          display="flex"
        >
          <div
            className="sc-AxjAm deKSuC"
            width={
              Object {
                "_": 40,
                "tablet": 48,
              }
            }
          >
            <div
              className=" gatsby-image-wrapper"
              style={
                Object {
                  "maxHeight": null,
                  "maxWidth": null,
                  "overflow": "hidden",
                  "position": "relative",
                }
              }
            >
              <div
                aria-hidden={true}
                style={
                  Object {
                    "paddingBottom": "65.5%",
                    "width": "100%",
                  }
                }
              />
              <img
                alt=""
                aria-hidden={true}
                src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAANCAYAAACpUE5eAAAACXBIWXMAAA3XAAAN1wFCKJt4AAAAkklEQVQ4y+2TsQ6CMBRF+1VODDQmJSJuNP6Lg7uTkwv4H/4XfQSMzbu00hpnYIPh5G4nZ3hPSFW1MquRqor97g9P7JIHLtcXrO1hDIGI0HqMAVmLz+0OyAJ8OgNH7Sh5XN0IL3Fw2IlCR66/UpGqIJtVqGMhx0LMLgxswrUIefyUetE75AUKORa+/6WThPnvl7sBgGbplMv6i1kAAAAASUVORK5CYII="
                style={
                  Object {
                    "borderRadius": 4,
                    "height": "100%",
                    "left": 0,
                    "objectFit": "cover",
                    "objectPosition": "center",
                    "opacity": 1,
                    "position": "absolute",
                    "top": 0,
                    "transitionDelay": "500ms",
                    "width": "100%",
                  }
                }
              />
              <picture>
                <source
                  sizes="(max-width: 512px) 100vw, 512px"
                  srcSet="/static/4ab94476b33b6da83177b2ad7f0aa6cd/69585/flag_france.png 200w,
/static/4ab94476b33b6da83177b2ad7f0aa6cd/497c6/flag_france.png 400w,
/static/4ab94476b33b6da83177b2ad7f0aa6cd/bc59e/flag_france.png 512w"
                />
                <img
                  alt=""
                  loading="lazy"
                  onLoad={[Function]}
                  sizes="(max-width: 512px) 100vw, 512px"
                  src="/static/4ab94476b33b6da83177b2ad7f0aa6cd/bc59e/flag_france.png"
                  srcSet="/static/4ab94476b33b6da83177b2ad7f0aa6cd/69585/flag_france.png 200w,
/static/4ab94476b33b6da83177b2ad7f0aa6cd/497c6/flag_france.png 400w,
/static/4ab94476b33b6da83177b2ad7f0aa6cd/bc59e/flag_france.png 512w"
                  style={
                    Object {
                      "borderRadius": 4,
                      "height": "100%",
                      "left": 0,
                      "objectFit": "cover",
                      "objectPosition": "center",
                      "opacity": 0,
                      "position": "absolute",
                      "top": 0,
                      "transition": "opacity 500ms",
                      "width": "100%",
                    }
                  }
                />
              </picture>
              <noscript
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<picture><source srcset=\\"/static/4ab94476b33b6da83177b2ad7f0aa6cd/69585/flag_france.png 200w,
/static/4ab94476b33b6da83177b2ad7f0aa6cd/497c6/flag_france.png 400w,
/static/4ab94476b33b6da83177b2ad7f0aa6cd/bc59e/flag_france.png 512w\\" sizes=\\"(max-width: 512px) 100vw, 512px\\" /><img loading=\\"lazy\\" sizes=\\"(max-width: 512px) 100vw, 512px\\" srcset=\\"/static/4ab94476b33b6da83177b2ad7f0aa6cd/69585/flag_france.png 200w,
/static/4ab94476b33b6da83177b2ad7f0aa6cd/497c6/flag_france.png 400w,
/static/4ab94476b33b6da83177b2ad7f0aa6cd/bc59e/flag_france.png 512w\\" src=\\"/static/4ab94476b33b6da83177b2ad7f0aa6cd/bc59e/flag_france.png\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
                  }
                }
              />
            </div>
          </div>
          <div
            className="sc-AxjAm jqJTTW"
            display="flex"
          >
            <div
              className="sc-AxjAm UbozS"
              height={
                Object {
                  "_": 12,
                  "tablet": 16,
                }
              }
            >
              <svg
                className="sc-AxjAm eQlzOq"
                height="100%"
                viewBox="0 0 18 18"
              >
                <path
                  d="M9.5 14.25l-5.584 2.936 1.066-6.218L.465 6.564l6.243-.907L9.5 0l2.792 5.657 6.243.907-4.517 4.404 1.066 6.218"
                  fill="white"
                />
              </svg>
            </div>
            <div
              className="sc-AxjAm UbozS"
              height={
                Object {
                  "_": 12,
                  "tablet": 16,
                }
              }
            >
              <svg
                className="sc-AxjAm eQlzOq"
                height="100%"
                viewBox="0 0 18 18"
              >
                <path
                  d="M9.5 14.25l-5.584 2.936 1.066-6.218L.465 6.564l6.243-.907L9.5 0l2.792 5.657 6.243.907-4.517 4.404 1.066 6.218"
                  fill="white"
                />
              </svg>
            </div>
            <div
              className="sc-AxjAm UbozS"
              height={
                Object {
                  "_": 12,
                  "tablet": 16,
                }
              }
            >
              <svg
                className="sc-AxjAm eQlzOq"
                height="100%"
                viewBox="0 0 18 18"
              >
                <path
                  d="M9.5 14.25l-5.584 2.936 1.066-6.218L.465 6.564l6.243-.907L9.5 0l2.792 5.657 6.243.907-4.517 4.404 1.066 6.218"
                  fill="white"
                />
              </svg>
            </div>
            <div
              className="sc-AxjAm UbozS"
              height={
                Object {
                  "_": 12,
                  "tablet": 16,
                }
              }
            >
              <svg
                className="sc-AxjAm eQlzOq"
                height="100%"
                viewBox="0 0 18 18"
              >
                <path
                  d="M9.5 14.25l-5.584 2.936 1.066-6.218L.465 6.564l6.243-.907L9.5 0l2.792 5.657 6.243.907-4.517 4.404 1.066 6.218"
                  fill="white"
                />
              </svg>
            </div>
            <div
              className="sc-AxjAm UbozS"
              height={
                Object {
                  "_": 12,
                  "tablet": 16,
                }
              }
            >
              <svg
                className="sc-AxjAm eQlzOq"
                height="100%"
                viewBox="0 0 18 18"
              >
                <path
                  d="M9.5 14.25l-5.584 2.936 1.066-6.218L.465 6.564l6.243-.907L9.5 0l2.792 5.657 6.243.907-4.517 4.404 1.066 6.218"
                  fill="white"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          className="sc-AxjAm ftcyfC"
          display="flex"
        >
          <div
            className="sc-AxjAm deKSuC"
            width={
              Object {
                "_": 40,
                "tablet": 48,
              }
            }
          >
            <div
              className=" gatsby-image-wrapper"
              style={
                Object {
                  "maxHeight": null,
                  "maxWidth": null,
                  "overflow": "hidden",
                  "position": "relative",
                }
              }
            >
              <div
                aria-hidden={true}
                style={
                  Object {
                    "paddingBottom": "65.5%",
                    "width": "100%",
                  }
                }
              />
              <img
                alt=""
                aria-hidden={true}
                src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAANCAYAAACpUE5eAAAACXBIWXMAAA3XAAAN1wFCKJt4AAABvUlEQVQ4y6WS204aURSG97t5aPEMHhjQHkbRwNARGKxWe9X0LVoj1mNQOfSiXpjY2F4bFRlAi9rUJ2hSwlAL63fNYJtY42D04suXvZL95187W3T0xIzWtlm0dc1Rs+M9tffEqOnRO+I5NbNbnAv0xRUleJ7QH/cwkVu+iSTXIMkgafiXkJ6uIjCeIr+awqiSgBJKwxdIIMj2v0hiyLeGzcWvwPciStkCyvmbGHVTJX8E4XIvEgdyyAaNBRPEoXXzWVGT1C+v05b8lqC9pIvQK6qGpm4D1fAUiT7PElTtIwLcUAmlzLb/PM5zz/M4Nue2gYKO0u4BjD17rEC/msQor2muPKbUbRIIJjAwsgFuCG4Ibmi2sEX0e7lhJM0NkxYKN/trNZyGV47j0/wOcHKMUkZHOZuzRfQNrqC9dwFOaQkO5wf0epfxuHse5tzRHUPnYByfn70Bp+NCmUAtGLVFnBXOcZL7gSJzmj9HUb/ub+yf+hEqmSzKhzqMBgiEJ4HIf4SvmyL8dpHpOyHKu/tohGGxd2V7RDX6mhhc+SFYGaJS/z/U6H/dATKzRFWb+W3tr00T+37w3ZrGGdGZyiUR4g0/JICeugAAAABJRU5ErkJggg=="
                style={
                  Object {
                    "borderRadius": 4,
                    "height": "100%",
                    "left": 0,
                    "objectFit": "cover",
                    "objectPosition": "center",
                    "opacity": 1,
                    "position": "absolute",
                    "top": 0,
                    "transitionDelay": "500ms",
                    "width": "100%",
                  }
                }
              />
              <picture>
                <source
                  sizes="(max-width: 512px) 100vw, 512px"
                  srcSet="/static/940a9bffd9cca715ab6601a2b6504e1b/69585/flag_usa.png 200w,
/static/940a9bffd9cca715ab6601a2b6504e1b/497c6/flag_usa.png 400w,
/static/940a9bffd9cca715ab6601a2b6504e1b/bc59e/flag_usa.png 512w"
                />
                <img
                  alt=""
                  loading="lazy"
                  onLoad={[Function]}
                  sizes="(max-width: 512px) 100vw, 512px"
                  src="/static/940a9bffd9cca715ab6601a2b6504e1b/bc59e/flag_usa.png"
                  srcSet="/static/940a9bffd9cca715ab6601a2b6504e1b/69585/flag_usa.png 200w,
/static/940a9bffd9cca715ab6601a2b6504e1b/497c6/flag_usa.png 400w,
/static/940a9bffd9cca715ab6601a2b6504e1b/bc59e/flag_usa.png 512w"
                  style={
                    Object {
                      "borderRadius": 4,
                      "height": "100%",
                      "left": 0,
                      "objectFit": "cover",
                      "objectPosition": "center",
                      "opacity": 0,
                      "position": "absolute",
                      "top": 0,
                      "transition": "opacity 500ms",
                      "width": "100%",
                    }
                  }
                />
              </picture>
              <noscript
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<picture><source srcset=\\"/static/940a9bffd9cca715ab6601a2b6504e1b/69585/flag_usa.png 200w,
/static/940a9bffd9cca715ab6601a2b6504e1b/497c6/flag_usa.png 400w,
/static/940a9bffd9cca715ab6601a2b6504e1b/bc59e/flag_usa.png 512w\\" sizes=\\"(max-width: 512px) 100vw, 512px\\" /><img loading=\\"lazy\\" sizes=\\"(max-width: 512px) 100vw, 512px\\" srcset=\\"/static/940a9bffd9cca715ab6601a2b6504e1b/69585/flag_usa.png 200w,
/static/940a9bffd9cca715ab6601a2b6504e1b/497c6/flag_usa.png 400w,
/static/940a9bffd9cca715ab6601a2b6504e1b/bc59e/flag_usa.png 512w\\" src=\\"/static/940a9bffd9cca715ab6601a2b6504e1b/bc59e/flag_usa.png\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
                  }
                }
              />
            </div>
          </div>
          <div
            className="sc-AxjAm jqJTTW"
            display="flex"
          >
            <div
              className="sc-AxjAm UbozS"
              height={
                Object {
                  "_": 12,
                  "tablet": 16,
                }
              }
            >
              <svg
                className="sc-AxjAm eQlzOq"
                height="100%"
                viewBox="0 0 18 18"
              >
                <path
                  d="M9.5 14.25l-5.584 2.936 1.066-6.218L.465 6.564l6.243-.907L9.5 0l2.792 5.657 6.243.907-4.517 4.404 1.066 6.218"
                  fill="white"
                />
              </svg>
            </div>
            <div
              className="sc-AxjAm UbozS"
              height={
                Object {
                  "_": 12,
                  "tablet": 16,
                }
              }
            >
              <svg
                className="sc-AxjAm eQlzOq"
                height="100%"
                viewBox="0 0 18 18"
              >
                <path
                  d="M9.5 14.25l-5.584 2.936 1.066-6.218L.465 6.564l6.243-.907L9.5 0l2.792 5.657 6.243.907-4.517 4.404 1.066 6.218"
                  fill="white"
                />
              </svg>
            </div>
            <div
              className="sc-AxjAm UbozS"
              height={
                Object {
                  "_": 12,
                  "tablet": 16,
                }
              }
            >
              <svg
                className="sc-AxjAm eQlzOq"
                height="100%"
                viewBox="0 0 18 18"
              >
                <path
                  d="M9.5 14.25l-5.584 2.936 1.066-6.218L.465 6.564l6.243-.907L9.5 0l2.792 5.657 6.243.907-4.517 4.404 1.066 6.218"
                  fill="white"
                />
              </svg>
            </div>
            <div
              className="sc-AxjAm UbozS"
              height={
                Object {
                  "_": 12,
                  "tablet": 16,
                }
              }
            >
              <svg
                className="sc-AxjAm eQlzOq"
                height="100%"
                viewBox="0 0 18 18"
              >
                <path
                  d="M9.5 14.25l-5.584 2.936 1.066-6.218L.465 6.564l6.243-.907L9.5 0l2.792 5.657 6.243.907-4.517 4.404 1.066 6.218"
                  fill="white"
                />
              </svg>
            </div>
            <div
              className="sc-AxjAm UbozS"
              height={
                Object {
                  "_": 12,
                  "tablet": 16,
                }
              }
            >
              <svg
                className="sc-AxjAm eQlzOq"
                height="100%"
                viewBox="0 0 18 18"
              >
                <g>
                  <path
                    d="M9.5 14.25l-5.584 2.936 1.066-6.218L.465 6.564l6.243-.907L9.5 0l2.792 5.657 6.243.907-4.517 4.404 1.066 6.218"
                    fill="grey"
                  />
                  <path
                    d="M9.5,14.25L3.9160000000000004,17.186L4.982,10.968L0.465,6.564L6.708,5.657L9.5,0"
                    fill="white"
                  />
                </g>
              </svg>
            </div>
          </div>
        </div>
        <div
          className="sc-AxjAm ftcyfC"
          display="flex"
        >
          <div
            className="sc-AxjAm deKSuC"
            width={
              Object {
                "_": 40,
                "tablet": 48,
              }
            }
          >
            <div
              className=" gatsby-image-wrapper"
              style={
                Object {
                  "maxHeight": null,
                  "maxWidth": null,
                  "overflow": "hidden",
                  "position": "relative",
                }
              }
            >
              <div
                aria-hidden={true}
                style={
                  Object {
                    "paddingBottom": "65.5%",
                    "width": "100%",
                  }
                }
              />
              <img
                alt=""
                aria-hidden={true}
                src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAANCAYAAACpUE5eAAAACXBIWXMAAA3XAAAN1wFCKJt4AAABhUlEQVQ4y42Tu0oDQRSG982SoKAmMWLwiqWFjUUqsdFKwULEyoewsLBQRBtRLAQb8YqgxCcIm7nsdX7nzGTX3WQTUvzMcGb45j/nzHEYY1wIAb0qzjmMGDNiyb7btcrGSMl9zpWUklbXIRgFCmEaQisLQ7BIi+72YkVQYjkZZ2oAFkXgrgvx9gHx+g7e6YDFURE0ZTiZF/pgIeTjE+KNFjDdBKbmEa9vQt49mLMcNMPIAVNYEEB8fkHNrQKVOlR9yQgTs0B1AfL5xZQhhQ4DJg1gKkZwfAKUZgxU1Rat6IFSFeH+obnD3e5oh2kNPQ/R1i4w2bDuEiC51LGotT1eyqlDXfjw4Ago16AaKyZNktnrEkQ7e2kdxwPq7lJDyA01g0CqsWybU6nBu73Xj9qUR9aQ5aAhvPMLqOaacYpy1dTQPz3TDQn+3RU4zP3D7P+i1EX7F97VDbzLa4jvH/sP+ycp+w+LJiXn1PfAACvfHxzB/knRGz91OGT8aFqMhowdTVvPmPwDhlG0EBO+DacAAAAASUVORK5CYII="
                style={
                  Object {
                    "borderRadius": 4,
                    "height": "100%",
                    "left": 0,
                    "objectFit": "cover",
                    "objectPosition": "center",
                    "opacity": 1,
                    "position": "absolute",
                    "top": 0,
                    "transitionDelay": "500ms",
                    "width": "100%",
                  }
                }
              />
              <picture>
                <source
                  sizes="(max-width: 512px) 100vw, 512px"
                  srcSet="/static/c5bc9cc27f496ada75b2f98d7ea04ad1/69585/flag_japan.png 200w,
/static/c5bc9cc27f496ada75b2f98d7ea04ad1/497c6/flag_japan.png 400w,
/static/c5bc9cc27f496ada75b2f98d7ea04ad1/bc59e/flag_japan.png 512w"
                />
                <img
                  alt=""
                  loading="lazy"
                  onLoad={[Function]}
                  sizes="(max-width: 512px) 100vw, 512px"
                  src="/static/c5bc9cc27f496ada75b2f98d7ea04ad1/bc59e/flag_japan.png"
                  srcSet="/static/c5bc9cc27f496ada75b2f98d7ea04ad1/69585/flag_japan.png 200w,
/static/c5bc9cc27f496ada75b2f98d7ea04ad1/497c6/flag_japan.png 400w,
/static/c5bc9cc27f496ada75b2f98d7ea04ad1/bc59e/flag_japan.png 512w"
                  style={
                    Object {
                      "borderRadius": 4,
                      "height": "100%",
                      "left": 0,
                      "objectFit": "cover",
                      "objectPosition": "center",
                      "opacity": 0,
                      "position": "absolute",
                      "top": 0,
                      "transition": "opacity 500ms",
                      "width": "100%",
                    }
                  }
                />
              </picture>
              <noscript
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<picture><source srcset=\\"/static/c5bc9cc27f496ada75b2f98d7ea04ad1/69585/flag_japan.png 200w,
/static/c5bc9cc27f496ada75b2f98d7ea04ad1/497c6/flag_japan.png 400w,
/static/c5bc9cc27f496ada75b2f98d7ea04ad1/bc59e/flag_japan.png 512w\\" sizes=\\"(max-width: 512px) 100vw, 512px\\" /><img loading=\\"lazy\\" sizes=\\"(max-width: 512px) 100vw, 512px\\" srcset=\\"/static/c5bc9cc27f496ada75b2f98d7ea04ad1/69585/flag_japan.png 200w,
/static/c5bc9cc27f496ada75b2f98d7ea04ad1/497c6/flag_japan.png 400w,
/static/c5bc9cc27f496ada75b2f98d7ea04ad1/bc59e/flag_japan.png 512w\\" src=\\"/static/c5bc9cc27f496ada75b2f98d7ea04ad1/bc59e/flag_japan.png\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
                  }
                }
              />
            </div>
          </div>
          <div
            className="sc-AxjAm jqJTTW"
            display="flex"
          >
            <div
              className="sc-AxjAm UbozS"
              height={
                Object {
                  "_": 12,
                  "tablet": 16,
                }
              }
            >
              <svg
                className="sc-AxjAm eQlzOq"
                height="100%"
                viewBox="0 0 18 18"
              >
                <path
                  d="M9.5 14.25l-5.584 2.936 1.066-6.218L.465 6.564l6.243-.907L9.5 0l2.792 5.657 6.243.907-4.517 4.404 1.066 6.218"
                  fill="white"
                />
              </svg>
            </div>
            <div
              className="sc-AxjAm UbozS"
              height={
                Object {
                  "_": 12,
                  "tablet": 16,
                }
              }
            >
              <svg
                className="sc-AxjAm eQlzOq"
                height="100%"
                viewBox="0 0 18 18"
              >
                <path
                  d="M9.5 14.25l-5.584 2.936 1.066-6.218L.465 6.564l6.243-.907L9.5 0l2.792 5.657 6.243.907-4.517 4.404 1.066 6.218"
                  fill="white"
                />
              </svg>
            </div>
            <div
              className="sc-AxjAm UbozS"
              height={
                Object {
                  "_": 12,
                  "tablet": 16,
                }
              }
            >
              <svg
                className="sc-AxjAm eQlzOq"
                height="100%"
                viewBox="0 0 18 18"
              >
                <path
                  d="M9.5 14.25l-5.584 2.936 1.066-6.218L.465 6.564l6.243-.907L9.5 0l2.792 5.657 6.243.907-4.517 4.404 1.066 6.218"
                  fill="white"
                />
              </svg>
            </div>
            <div
              className="sc-AxjAm UbozS"
              height={
                Object {
                  "_": 12,
                  "tablet": 16,
                }
              }
            >
              <svg
                className="sc-AxjAm eQlzOq"
                height="100%"
                viewBox="0 0 18 18"
              >
                <path
                  d="M9.5 14.25l-5.584 2.936 1.066-6.218L.465 6.564l6.243-.907L9.5 0l2.792 5.657 6.243.907-4.517 4.404 1.066 6.218"
                  fill="grey"
                />
              </svg>
            </div>
            <div
              className="sc-AxjAm UbozS"
              height={
                Object {
                  "_": 12,
                  "tablet": 16,
                }
              }
            >
              <svg
                className="sc-AxjAm eQlzOq"
                height="100%"
                viewBox="0 0 18 18"
              >
                <path
                  d="M9.5 14.25l-5.584 2.936 1.066-6.218L.465 6.564l6.243-.907L9.5 0l2.792 5.657 6.243.907-4.517 4.404 1.066 6.218"
                  fill="grey"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="sc-AxjAm ddUQGN"
      display={
        Object {
          "_": "block",
          "tablet": "none",
        }
      }
      height={64}
      width={64}
    >
      <button
        className="sc-AxjAm sc-AxirZ bydKTu"
        height="100%"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        width="100%"
      >
        <svg
          className="sc-AxiKw jxzaqN"
          fill="white"
          height={24}
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </button>
    </div>
  </header>,
]
`;
